pkg/build/controller/buildconfig/buildconfig_controller.go

```diff
@@ -174,9 +176,55 @@ func (c *BuildConfigController) buildConfigAdded(obj interface{}) {
 // buildconfig is updated or there is a relist of buildconfigs
 func (c *BuildConfigController) buildConfigUpdated(old, cur interface{}) {
 	bc := cur.(*buildv1.BuildConfig)
+	oldBuildConfig := old.(*buildv1.BuildConfig)
+	// TODO: Remove the counter and event after the image trigger behavior is removed.
+	if c.imageChangeTriggerCleared(oldBuildConfig, bc) {
+		prometheus.MarkImageChangeTriggerCleared(bc.Namespace, bc.Name, true)
+		c.recorder.Event(bc,
+			corev1.EventTypeWarning,
+			"ImageChangeTriggerCleared",
+			"spec.triggers[*].imagechange.lastTriggeredImageID was cleared, which will trigger a build. This behavior is deprecated and will be removed in a future OpenShift release.")
+	} else {
+		prometheus.MarkImageChangeTriggerCleared(bc.Namespace, bc.Name, false)
```

Comment:
Perhaps we can be a bit more intelligent and only mark cleared to `false` if:

1. BuildConfig has a "valid" ImageChange trigger (unchanged, updates from empty -> sha, or sha -> sha)
2. BuildConfig previously had an ImageChange trigger, but it was removed.

In a real cluster, this would significantly cut down the time series size.